src/sentry/workflow_engine/models/workflow.py

```diff
@@ -67,16 +68,26 @@ class Meta:
             )
         ]
 
-    def evaluate_trigger_conditions(self, job: WorkflowJob) -> bool:
+    def evaluate_trigger_conditions(self, job: DataJob) -> bool:
         """
         Evaluate the conditions for the workflow trigger and return if the evaluation was successful.
         If there aren't any workflow trigger conditions, the workflow is considered triggered.
         """
         if self.when_condition_group is None:
             return True
 
-        job["workflow"] = self
-        evaluation, _ = evaluate_condition_group(self.when_condition_group, job)
+        if evaluate_all_conditions := isinstance(job.get("event"), GroupEvent):
+            job["workflow"] = self
+            evaluation, _ = evaluate_condition_group(
+                self.when_condition_group,
+                cast(WorkflowJob, job),
+                evaluate_slow_conditions=not evaluate_all_conditions,
+            )
+        else:
+            evaluation, _ = evaluate_condition_group(
+                self.when_condition_group, job, evaluate_slow_conditions=not evaluate_all_conditions
+            )
```

Comment:
this either passes `DataJob` or `WorkflowJob` to `evaluate_condition_group`, not the union type